system-design/class-diagram/domain/validator.puml

```diff
@@ -0,0 +1,40 @@
+@startuml
+package validator {
+    package exceptions {
+        enum SeverityLevel {
+            Low,
+            Medium,
+            High,
+            Critical
+        }
+
+        class CustomException {
+            + Severity: SeverityLevel
+            + ResolutionSuggestion: string
+        }
+        CustomException <|-- PatronException
+        CustomException <|-- BookException
+        CustomException --( Exception
+        CustomException *-- SeverityLevel
```

Comment:
I think that this is not composition, is just "Directed association".

Reply:
I could change it to an aggregate relationship, given that `SecurityLevel` can be used for logging.